modules/ingest-otel/src/main/java/org/elasticsearch/ingest/otel/NormalizeForStreamProcessor.java

```diff
@@ -103,6 +108,41 @@ public IngestDocument execute(IngestDocument document) {
 
         // non-OTel document
 
+        // handling structured messages
+        Map<String, Object> body = null;
+        try {
+            String message = document.getFieldValue("message", String.class, true);
+            if (message != null) {
+                message = message.trim();
+                if (message.startsWith("{") && message.endsWith("}")) {
+                    // if the message is a JSON object, we assume it is a structured log
+                    Object parsedMessage = JsonProcessor.apply(message, true, true);
+                    if (parsedMessage instanceof Map) {
+                        @SuppressWarnings("unchecked")
+                        Map<String, Object> messageMap = (Map<String, Object>) parsedMessage;
+                        if (messageMap.containsKey("@timestamp")) {
+                            log.debug(
+                                "Handling structured message with @timestamp field, assuming ECS-JSON format, merging into root document"
+                            );
+                            source.remove("message");
+                            JsonProcessor.recursiveMerge(source, messageMap);
+                        } else {
+                            log.debug(
+                                "Handling structured message without @timestamp field, assuming non-ECS format, moving to 'body.structured'"
+                            );
+                            body = new HashMap<>();
+                            body.put(STRUCTURED_KEY, messageMap);
+                            source.remove("message");
+                        }
+                    } else {
+                        log.debug("Structured message is not a JSON object, keeping it as a string in 'body.text' field: {}", message);
```

Comment:
Are we happy this is going to be a relatively rare event based on the checks we have (startsWith & endsWith)?

I have a feeling that the JsonProcessor is likely to be pretty heavy on CPU and RAM so I'm curious on the trade off of more pre-checks (Counting `{`'s vs `}`'s etc) vs less checks and saving CPU cycles there if it's a scenario we rarely see but that depends heavily on our anticipated data patterns.

Reply:
The assumption is:
* Doesn't start/end with {/}: extremely common (98%)
* Start/end with {/}, and is JSON: rare, but not crazy rare (~2%)
* Start/end with {/}, and is not JSON: very rare

I don't have a good feeling how expensive it is to invoke the processor vs. doing pre-checks, but I think this is an easy way to get us going, we can improve performance later on if we notice that this is a significant issue.

Reply:
OK sounds like this is optimized for the common case already then 👍🏻